### PR TITLE
Adds docs: security-guidelines and blueprinting-tips

### PR DESCRIPTION
**Pull request by aledsage**
And removes guide/dev/rest/, because that is covered by docs/guide/ops/rest.md.

### CODE REVIEW

File: docs/guide/yaml/blueprinting-tips.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* Some entities require `sudo` rights, which must be granted to the user running Brooklyn.
+
+
+### Deploying to Bring Your Own Nodes (BYON)
+
+A BYON location can be defined, which avoids the time required to provision VMs. This is fast,
+but has the downside that artifacts installed during a previous run can interfere with subsequent 
+runs.
+
+A variant of this is to use Vagrant (e.g. with VirtualBox) to create VMs on your local machine,
```
Comment (sjcorbett) [inline docs/guide/yaml/blueprinting-tips.md]: Could reference the jclouds-vagrant provider written by @neykov.

**Comment by sjcorbett** [comment]
Looks good. A "see also" section at the end linking to docs on entitlements and external configuration (assuming we have them) might be useful too.

File: docs/guide/yaml/blueprinting-tips.md (hunk 1 above)
Comment (ahgittin) [inline docs/guide/yaml/blueprinting-tips.md]: In fact we should reference the Getting Started Guide improvements @johnmccabe on joining which uses this, i believe?

**Comment by ahgittin** [comment]
useful doc improvements.  good to merge, although addressing the comments or sticking a `TODO` in the docs might be wise first!

**Comment by hzbarcea** [comment]
I created [BROOKLYN-198](https://issues.apache.org/jira/browse/BROOKLYN-198) as a placeholder for comments that should be addressed. I assume it's good enough to have only one (or a few) such placeholders for a release. Once comments to a PR have been addressed in a follow-up commit we could edit the issue and cross over the PR addressed.

That said, tested, merging...

**Pull request merged**